"""Correct act_cats table which had primary key on wrong column.

Revision ID: 2b8dfc406f65
Revises: 012578ec09ec
Create Date: 2019-04-30 14:46:22.268455

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '2b8dfc406f65'
down_revision = '012578ec09ec'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('act_cats',
    sa.Column('a_done', sa.String(length=40), nullable=False),
    sa.Column('a_cat', sa.String(length=40), nullable=True),
    sa.PrimaryKeyConstraint('a_done')
    )
    op.create_table('act_recs',
    sa.Column('a_id', sa.Integer(), nullable=False),
    sa.Column('day', sa.Date(), nullable=True),
    sa.Column('startt', sa.TIME(), nullable=True),
    sa.Column('endt', sa.TIME(), nullable=True),
    sa.Column('a_done', sa.String(length=40), nullable=True),
    sa.Column('comments', sa.String(length=50), nullable=True),
    sa.ForeignKeyConstraint(['a_done'], ['act_cats.a_done'], ),
    sa.PrimaryKeyConstraint('a_id'),
    sa.UniqueConstraint('day', 'endt', name='uniq_endt'),
    sa.UniqueConstraint('day', 'startt', name='uniq_startt')
    )
    op.create_index(op.f('ix_act_recs_a_done'), 'act_recs', ['a_done'], unique=False)
    op.create_index(op.f('ix_act_recs_day'), 'act_recs', ['day'], unique=False)
    op.create_index(op.f('ix_act_recs_startt'), 'act_recs', ['startt'], unique=False)
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_index(op.f('ix_act_recs_startt'), table_name='act_recs')
    op.drop_index(op.f('ix_act_recs_day'), table_name='act_recs')
    op.drop_index(op.f('ix_act_recs_a_done'), table_name='act_recs')
    op.drop_table('act_recs')
    op.drop_table('act_cats')
    # ### end Alembic commands ###
